src/Text.Analyzers/Core/CodeAnalysisDictionary.cs

```diff
@@ -0,0 +1,134 @@
+
+        /// <summary>
+        /// Creates a new instance of this class with recognized and unrecognized words (if specified) loaded
+        /// from the specified XML <paramref name="streamReader"/>.
+        /// </summary>
+        /// <param name="streamReader">XML stream of a code analysis dictionary.</param>
+        /// <returns>A new instance of this class with the words loaded.</returns>
+        public static CodeAnalysisDictionary CreateFromXml(StreamReader streamReader)
+        {
+            var document = XDocument.Load(streamReader);
+            return new CodeAnalysisDictionary(document);
+        }
+
+        /// <summary>
+        /// Creates a new instance of this class with recognized words loaded from the specified DIC <paramref name="streamReader"/>.
+        /// </summary>
+        /// <param name="streamReader">DIC stream of a code analysis dictionary.</param>
+        /// <returns>A new instance of this class with recognized words loaded.</returns>
+        public static CodeAnalysisDictionary CreateFromDic(StreamReader streamReader)
+        {
+            var words = new Collection<string>();
+
+            string word;
+            while ((word = streamReader.ReadLine()) != null)
+            {
+                if (!string.IsNullOrWhiteSpace(word.Trim()))
+                {
+                    words.Add(word);
+                }
+            }
+
+            return new CodeAnalysisDictionary(words);
+        }
+
+        /// <summary>
+        /// Returns a copy of the current instance.
+        /// </summary>
+        /// <returns></returns>
+        public CodeAnalysisDictionary Clone() => new CodeAnalysisDictionary(this);
+
+        /// <summary>
+        /// Replaces this instance's <see cref="RecognizedWords"/> and <see cref="UnrecognizedWords"/> with
+        /// the union of its words and <paramref name="other"/>'s words.
+        /// </summary>
+        /// <param name="other">Another instance of this class.</param>
+        /// <returns>This instance with words added from the other instance.</returns>
+        public CodeAnalysisDictionary CombineWith(CodeAnalysisDictionary other)
+        {
+            RecognizedWords.UnionWith(other.RecognizedWords);
+            UnrecognizedWords.UnionWith(other.UnrecognizedWords);
+            return this;
+        }
+
+        private static IEnumerable<T> GetValues<T>(IEnumerable<XElement> elements, Func<XElement, T> extractor)
+            => elements.Select(extractor);
+
+        private static IEnumerable<XElement> GetElements(XDocument document, string section, string name)
+            => document.Descendants(section).SelectMany(x => x.Elements(name));
+
+        private static string ExtractInnerText(XElement element) => element.Value.Trim();
+
+        private void LoadWordsFromDocument(XDocument document)
+        {
+            RecognizedWords.UnionWith(GetWordsOrExceptions("Recognized", "Word"));
```

Comment:
Probably add a TODO comment here to handle words marked as Deprecated and handling of Acronyms as per the documentation here: https://docs.microsoft.com/visualstudio/code-quality/ca1704?view=vs-2019#to-add-words-to-a-custom-dictionary

Reply:
Done